prov/usnic/src/usdf_fabric.c

```diff
@@ -341,6 +341,94 @@ static int validate_modebits(uint32_t version, const struct fi_info *hints,
 	return FI_SUCCESS;
 }
 
+static int usdf_alloc_fid_nic(struct fi_info *fi,
+			struct usd_device_attrs *dap)
+{
+	int ret;
+	struct fid_nic *nic = NULL;
+	struct fi_device_attr *da = NULL;
+	struct fi_bus_attr *ba = NULL;
+	struct fi_link_attr *la = NULL;
+
+	nic = calloc(1, sizeof(*nic));
+	if (!nic)
+		goto nomem;
```

Comment:
The `nomem` label makes a call to `ofi_free_nic` if the `calloc` fails. With the current version of the code, that will cause a seg fault because `ofi_free_nic` is not safe if a `NULL` pointer is passed in.

Reply:
Fixed `ofi_free_nic()` to handle NULL.